mobile/src/main/java/org/openhab/habdroid/ui/ChartActivity.kt

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.habdroid.ui
+
+import android.content.Context
+import android.content.res.Configuration
+import android.os.Bundle
+import android.util.DisplayMetrics
+import android.util.Log
+import android.util.TypedValue
+import android.view.Menu
+import android.view.MenuItem
+import android.view.WindowManager
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import org.openhab.habdroid.R
+import org.openhab.habdroid.core.connection.ConnectionFactory
+import org.openhab.habdroid.model.Widget
+import org.openhab.habdroid.ui.widget.WidgetImageView
+import org.openhab.habdroid.util.ScreenLockMode
+import org.openhab.habdroid.util.getPrefs
+import org.openhab.habdroid.util.orDefaultIfEmpty
+import java.util.Random
+
+class ChartActivity : AbstractBaseActivity(), SwipeRefreshLayout.OnRefreshListener {
+    private lateinit var swipeLayout: SwipeRefreshLayout
+    private lateinit var chart: WidgetImageView
+    private lateinit var period: String
+    private lateinit var widget: Widget
+    private lateinit var chartTheme: CharSequence
+    private var density: Int = 0
+    private val random = Random()
+    private var showLegend: Boolean = true
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_chart)
+
+        widget = intent.getParcelableExtra(WIDGET)!!
+        period = widget.period
+        showLegend = widget.legend ?: true
```

Comment:
How about doing `!= false` instead of elvis? Or, even more clear, `widget != null && widget.legend`?

Reply:
`legend` itself can be `null`, so checking for `widget != null` doesn't help here. I find the elvis operator more clear in this case.
And IMO it's ok to crash if the widget is null.